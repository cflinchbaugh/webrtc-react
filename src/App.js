import React, { Component } from 'react';
import './App.css';
import TextArea from './components/inputs/TextArea';
import Button from './components/buttons/Button';
var Peer = require('simple-peer');

class App extends Component {
    constructor(props) {
        super(props);

        this.initializePeer = this.initializePeer.bind(this);
        this.handleReceiveMessage = this.handleReceiveMessage.bind(this);
        this.handleClickInitiatorButton = this.handleClickInitiatorButton.bind(this);
        this.handleClickNonInitiatorButton = this.handleClickNonInitiatorButton.bind(this);
        this.handleChangeYourId = this.handleChangeYourId.bind(this);
        this.handleChangeOtherId = this.handleChangeOtherId.bind(this);
        this.handleClickConnectButton = this.handleClickConnectButton.bind(this);
        this.handleChangeYourMessage = this.handleChangeYourMessage.bind(this);
        this.handleClickSendButton = this.handleClickSendButton.bind(this);
        
        this.state = {
            yourId: '',
            otherId: '',
            yourMessage: '',
            chatLog: ''
        }
    }

    render() {

        const initiatorButtonData ={
                id: 'initiator',
                label: 'Initiator',
                handleClickButton: this.handleClickInitiatorButton
            },
            nonInitiatorButtonData = {
                id: 'noninitiator',
                label: 'Non Initiator',
                handleClickButton: this.handleClickNonInitiatorButton
            },
            yourIdData = {
                id: 'your-id',
                value: this.state.yourId,
                handleChangeValue: this.handleChangeYourId
            },
            otherIdData = {
                id: 'other-id',
                value: this.state.otherId,
                handleChangeValue: this.handleChangeOtherId
            },
            connectButtonData = {
                id: 'connect',
                label: 'Connect',
                handleClickButton: this.handleClickConnectButton
            },
            yourMessageInputData = {
                id: 'your-message',
                value: this.state.yourMessage,
                handleChangeValue: this.handleChangeYourMessage
            },
            chatLogInputData = {
                id: 'chat-log',
                value: this.state.chatLog,
                handleChangeValue: () => {},
                disabled: true
            },
            sendMessageButtonData = {
                id: 'send-message',
                label: 'Send',
                handleClickButton: this.handleClickSendButton
            }

        return (
            <div className="App">
                <div>
                    <Button {...initiatorButtonData}/>
                    <Button {...nonInitiatorButtonData}/>
                </div>

                <div>
                    <label>Your Id:</label><br/>
                    <TextArea {...yourIdData} />
                </div>

                <div>
                    <label>Other Id:</label><br/>
                    <TextArea {...otherIdData}/>
                </div>

                <Button {...connectButtonData}/>

                <hr/>

                <div>
                    <label>Chat Log:</label><br/>
                    <TextArea {...chatLogInputData}/>
                </div>

                <div>
                    <label>Message:</label><br/>
                    <TextArea {...yourMessageInputData}/>
                </div>
                <Button {...sendMessageButtonData}/>
                
                <pre id="messages"></pre>
            </div>
        );
    }

    handleClickInitiatorButton() {
        this.setState({
            initiator: true
        });

        this.initializePeer(true)
    }

    handleClickNonInitiatorButton() {
        this.setState({
            initiator: false
        });

        this.initializePeer(false)
    }

    initializePeer(initiator) {
        const message = initiator ? 'Loading...' : 'Please Add Other Id';

        this.setState({
            yourId: message
        });

        this.peer = new Peer({
            initiator: initiator,
            trickle: false
        });

        this.peer.on('signal', (data) => {
            const yourId = JSON.stringify(data);
            this.setState({
                yourId: yourId
            });
        });

        this.peer.on('data', this.handleReceiveMessage);
    }

    handleReceiveMessage(data) {
        const updatedChatLog = this.state.chatLog += data;

        this.setState({
            chatLog: updatedChatLog
        });
    }

    handleChangeYourId(value) {
        this.setState({
            yourId: value
        });
    }

    handleChangeOtherId(value) {
        this.setState({
            otherId: value
        });
    }

    handleClickConnectButton() {
        this.peer.signal(this.state.otherId);
    }

    handleChangeYourMessage(value) {
        this.setState({
            yourMessage: value
        });
    }

    handleClickSendButton() {
        const updatedChatLog = this.state.chatLog += this.state.yourMessage;
        
        this.peer.send(this.state.yourMessage);

        this.setState({
            chatLog: updatedChatLog,
            yourMessage: ''
        });
    }
}

export default App;
