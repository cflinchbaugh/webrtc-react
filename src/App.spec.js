import React from 'react';
import { shallow } from 'enzyme';
import App from './App';


describe('App', () => {
    it('matches snapshot', () => {
        const data = {
           
        }
        
        expect(shallow(<App {...data}/>)).toMatchSnapshot();
    });

    
    // it('calls handleChangeValue prop when value changes', () => {
    //     const mockFunc = jest.fn();

    //     const data = {
    //         id: 'id',
    //         value: 'value',
    //         handleChangeValue: mockFunc
    //     }
    
        
    //     const component = shallow(<TextArea {...data}/>);

    //     const event = {
    //         currentTarget: {
    //             value: 'changed'
    //         }
    //     }

    //     component.find('textarea')
    //         .simulate('change', event);

    //     expect(mockFunc).toHaveBeenCalled();
    // });

    // it('calls handleChangeValue prop when value changes', () => {
    //     const mockFunc = jest.fn();

    //     const data = {
    //         id: 'id',
    //         value: 'value',
    //         handleChangeValue: mockFunc
    //     }
    
        
    //     const component = shallow(<TextArea {...data}/>);

    //     const event = {
    //         currentTarget: {
    //             value: 'changed'
    //         }
    //     }

    //     component.find('textarea')
    //         .simulate('change', event);

    //     expect(mockFunc).toHaveBeenCalledWith('changed');
    // });

});
