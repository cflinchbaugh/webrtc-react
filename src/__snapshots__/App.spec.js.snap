// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <Button
            disabled={false}
            handleClickButton={[Function]}
            id="initiator"
            label="Initiator"
          />
          <Button
            disabled={false}
            handleClickButton={[Function]}
            id="noninitiator"
            label="Non Initiator"
          />
        </div>,
        <div>
          <label>
            Your Id:
          </label>
          <br />
          <TextArea
            disabled={false}
            handleChangeValue={[Function]}
            id="your-id"
            value=""
          />
        </div>,
        <div>
          <label>
            Other Id:
          </label>
          <br />
          <TextArea
            disabled={false}
            handleChangeValue={[Function]}
            id="other-id"
            value=""
          />
        </div>,
        <Button
          disabled={true}
          handleClickButton={[Function]}
          id="connect"
          label="Connect"
        />,
        <hr />,
        <div>
          <label>
            Chat Log:
          </label>
          <br />
          <TextArea
            disabled={true}
            handleChangeValue={[Function]}
            id="chat-log"
            value=""
          />
        </div>,
        <div>
          <label>
            Message:
          </label>
          <br />
          <TextArea
            disabled={false}
            handleChangeValue={[Function]}
            id="your-message"
            value=""
          />
        </div>,
        <Button
          disabled={false}
          handleClickButton={[Function]}
          id="send-message"
          label="Send"
        />,
        <pre
          id="messages"
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Button
              disabled={false}
              handleClickButton={[Function]}
              id="initiator"
              label="Initiator"
            />,
            <Button
              disabled={false}
              handleClickButton={[Function]}
              id="noninitiator"
              label="Non Initiator"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disabled": false,
              "handleClickButton": [Function],
              "id": "initiator",
              "label": "Initiator",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disabled": false,
              "handleClickButton": [Function],
              "id": "noninitiator",
              "label": "Non Initiator",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Your Id:
            </label>,
            <br />,
            <TextArea
              disabled={false}
              handleChangeValue={[Function]}
              id="your-id"
              value=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Your Id:",
            },
            "ref": null,
            "rendered": "Your Id:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disabled": false,
              "handleChangeValue": [Function],
              "id": "your-id",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Other Id:
            </label>,
            <br />,
            <TextArea
              disabled={false}
              handleChangeValue={[Function]}
              id="other-id"
              value=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Other Id:",
            },
            "ref": null,
            "rendered": "Other Id:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disabled": false,
              "handleChangeValue": [Function],
              "id": "other-id",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "disabled": true,
          "handleClickButton": [Function],
          "id": "connect",
          "label": "Connect",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "hr",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Chat Log:
            </label>,
            <br />,
            <TextArea
              disabled={true}
              handleChangeValue={[Function]}
              id="chat-log"
              value=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Chat Log:",
            },
            "ref": null,
            "rendered": "Chat Log:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disabled": true,
              "handleChangeValue": [Function],
              "id": "chat-log",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Message:
            </label>,
            <br />,
            <TextArea
              disabled={false}
              handleChangeValue={[Function]}
              id="your-message"
              value=""
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Message:",
            },
            "ref": null,
            "rendered": "Message:",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "disabled": false,
              "handleChangeValue": [Function],
              "id": "your-message",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "disabled": false,
          "handleClickButton": [Function],
          "id": "send-message",
          "label": "Send",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "id": "messages",
        },
        "ref": null,
        "rendered": null,
        "type": "pre",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <Button
              disabled={false}
              handleClickButton={[Function]}
              id="initiator"
              label="Initiator"
            />
            <Button
              disabled={false}
              handleClickButton={[Function]}
              id="noninitiator"
              label="Non Initiator"
            />
          </div>,
          <div>
            <label>
              Your Id:
            </label>
            <br />
            <TextArea
              disabled={false}
              handleChangeValue={[Function]}
              id="your-id"
              value=""
            />
          </div>,
          <div>
            <label>
              Other Id:
            </label>
            <br />
            <TextArea
              disabled={false}
              handleChangeValue={[Function]}
              id="other-id"
              value=""
            />
          </div>,
          <Button
            disabled={true}
            handleClickButton={[Function]}
            id="connect"
            label="Connect"
          />,
          <hr />,
          <div>
            <label>
              Chat Log:
            </label>
            <br />
            <TextArea
              disabled={true}
              handleChangeValue={[Function]}
              id="chat-log"
              value=""
            />
          </div>,
          <div>
            <label>
              Message:
            </label>
            <br />
            <TextArea
              disabled={false}
              handleChangeValue={[Function]}
              id="your-message"
              value=""
            />
          </div>,
          <Button
            disabled={false}
            handleClickButton={[Function]}
            id="send-message"
            label="Send"
          />,
          <pre
            id="messages"
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Button
                disabled={false}
                handleClickButton={[Function]}
                id="initiator"
                label="Initiator"
              />,
              <Button
                disabled={false}
                handleClickButton={[Function]}
                id="noninitiator"
                label="Non Initiator"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "disabled": false,
                "handleClickButton": [Function],
                "id": "initiator",
                "label": "Initiator",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "disabled": false,
                "handleClickButton": [Function],
                "id": "noninitiator",
                "label": "Non Initiator",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Your Id:
              </label>,
              <br />,
              <TextArea
                disabled={false}
                handleChangeValue={[Function]}
                id="your-id"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Your Id:",
              },
              "ref": null,
              "rendered": "Your Id:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "disabled": false,
                "handleChangeValue": [Function],
                "id": "your-id",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Other Id:
              </label>,
              <br />,
              <TextArea
                disabled={false}
                handleChangeValue={[Function]}
                id="other-id"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Other Id:",
              },
              "ref": null,
              "rendered": "Other Id:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "disabled": false,
                "handleChangeValue": [Function],
                "id": "other-id",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": true,
            "handleClickButton": [Function],
            "id": "connect",
            "label": "Connect",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Chat Log:
              </label>,
              <br />,
              <TextArea
                disabled={true}
                handleChangeValue={[Function]}
                id="chat-log"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Chat Log:",
              },
              "ref": null,
              "rendered": "Chat Log:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "disabled": true,
                "handleChangeValue": [Function],
                "id": "chat-log",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <label>
                Message:
              </label>,
              <br />,
              <TextArea
                disabled={false}
                handleChangeValue={[Function]}
                id="your-message"
                value=""
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Message:",
              },
              "ref": null,
              "rendered": "Message:",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "disabled": false,
                "handleChangeValue": [Function],
                "id": "your-message",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "disabled": false,
            "handleClickButton": [Function],
            "id": "send-message",
            "label": "Send",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "id": "messages",
          },
          "ref": null,
          "rendered": null,
          "type": "pre",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
